uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-request-popup-details-tab.component.js

```diff
@@ -289,22 +284,12 @@ define([
     function daysSelectionModeChangeHandler () {
       vm.uiOptions.toDate = null;
 
-      resetUIDayTypesTimeAndDeductions('to');
+      resetUIInputs('to');
 
       return $q.resolve()
-        .then(function () {
-          if (isCalculationUnit('hours') && vm.uiOptions.fromDate) {
-            disableAndShowLoadingTimeInput('from');
-            (!vm.uiOptions.multipleDays) && disableAndShowLoadingTimeInput('to');
-
-            return loadTimeRangesFromWorkPattern('from')
-              .then(function () {
-                setDeductionMaximumBoundary('from', true);
-              })
-              .then(setRequestDateTimesAndDateTypes);
-          }
-        })
+        .then(setRequestDateTimesAndDateTypes)
         .then(vm.setDaysSelectionModeExtended)
+        .then(finishLoadingTimesAndDateTypes)
```

Comment:
isn't this supposed to go in a `.finally` statement?

Reply:
@reneolivo I don't think so, this func needs to fire **before** balance calculation.